### DISCUSSION TITLE
Using components with package prefix

### DISCUSSION THREAD

**Discussion by rabol** [wireui/wireui#539] [Q&A]
Hi

Is it possible to use WireUI components with package prefix like

<x-wireui::select ...>

**Comment by ph7jack** [comment]
Hi @rabol 
Yes, you can publish the wireui config and add a prefix to all components
1. Publish the config
`php artisan vendor:publish --tag="wireui.config"`
2. Rename each component alias with your prefix

**Reply by rabol** [reply]
I'm using PhpStorm and the Laravel IDEA extension, and they have added wireui support like this:

<x-wireui::button> but that gives errors

(https://github.com/laravel-idea/plugin/issues/630)

so how do i make the two things work together ?

**Reply by ph7jack** [reply]
Which errors?

**Reply by rabol** [reply]
eg. this:

```
        <div class="col-span-6">
            <x-wireui::input wire:model="user.name" label="Name" />
        </div>
```

gives this error:
Undefined variable $errorless

**Reply by ph7jack** [reply]
I see now, you are using wrong the wireui components.
See the documentation of their names https://livewire-wireui.com

Instead of
`<x-wireui::input wire:model="user.name" label="Name" />`
Use it
`<x-input wire:model="user.name" label="Name" />`

This `wireui::` is a namespace of the wireui views, but these views won't work without the component class
Use them without `wireui::`

**Reply by judgej** [reply]
Is the namespacing of blade components going to be something that the Laravel framework needs to do in its core?

For example, `<x-input>` is pretty generic, and all-so-easily clashes with blad components that other packages my introduce, or a bunch of hand-made components that were already in a project before WireUI was installed. It would be great just to namespace the components in some way, so `<x-input>` from another source would continue to work while `<wire-ui::x-input>` (or some other way of collecting the components together) gets switched in only where and when needed.

When the project already contains an `input` component, and then WireUI is installed, you never really know which is going to take precedence when `<x-input>` is used.

**Reply by judgej** [reply]
Oh, is this as simple as changing the `wireui.components` config keys? So if I change config `wireui.components.input.alias` from `input` to `wireui-input`, then I can use `<x-wireui-input>` in my views instead of `<x-input>`, which retains the functionality defined in application or another package? Do I feel stupid realising that is actually what you said further up the thread? Yes. Yes I do. :-)

**Comment by judgej** [comment]
@rabol does this help https://github.com/wireui/wireui/discussions/596 ? It assumes I've understood your original question, of course.

**Comment by Uhasith** [comment]
Thank you @judgej. You saved my Life  🥹
